### ISSUE TITLE
[4.x]: Disabling the default variant and saving a product breaks the product's page

### ISSUE THREAD

**Issue by Nellyaa** [craftcms/commerce#2891]
### Description

When you disable the one default variant a product has and save it, it breaks the product's page in the backend, rendering it inaccessible. I'm sorry I don't have the error at hand right now, but should be easy enough to reproduce.

### Steps to reproduce

1. Create a product type with multiple variants
2. Create a product with one variant
3. in the variant tab, disable the default variant from the gear menu so it gets the little red dot
4. Press save
5. It's now impossible to access the product's page anymore, exception is being thrown if attempted

### Expected behavior

The default variant either a) shouldn't be able to be disabled or b) it shouldn't break the product's page so you can still edit it

### Craft CMS version

4.0.5

### Craft Commerce version

4.0.4

### Installed plugins and versions

-

**Comment by robzor** [comment]
I've reported this and they have a fix but it hasn't been deployed.

I had this feedback from Dale at Craft:

We'll be releasing a fix for this bug on version 4.1 which will be released 1-2 weeks from now. however to get the fix early, change your `craftcms/commerce` requirement in composer.json to:

```json
"require": {
  "craftcms/commerce": "4.1.x-dev",
  "...": "..."
}
```

Then run `composer update`.

**Comment by lukeholder** [comment]
This is now fixed in 4.1 see changelog here: https://github.com/craftcms/commerce/blob/develop/CHANGELOG.md#410---2022-07-19
